Add --environ-file for defaults for env object

kanaka opened the pull request:
Use `--environ-file` option that parses and environment file in order to populate the base env object/map.

This uses version 1.24.0 of dotenvx (PR adding necessary functionality: https://github.com/dotenvx/dotenvx/issues/445)

Closely related to https://github.com/LonoCloud/conlink/pull/32

kanaka committed:
Add --environ-file default for env (using dotenvx).

If the --environ-file option is used then dotenvx is used to load and
parse the environment file to use a defaults env values.

Add an example suite examples/03-env.yaml that is designed to be used
with --environ-file examples/03-env-file.

Add tests to test/runexamples that tests examples/03-env.yaml with and
without the the examples/03-env-file.

NOTE: It would be preferable to have the option be named `--env-file`
since this is more common. However, when running in nbb with node
earlier than 22, node intercepts the --env-file option. And for some
reason, node refused absoluate paths with --env-file. In node v22 it is
possible to avoid the issue by using a "--" parameter before the
regular nbb script parameters. But node 22 is not in standard GHA runner
configs so for now we will use "--environ-file".

jonsmock reviewed (changes requested):
Following up on the actual PR with out-of-band discussion, looks good to me with the agreed upon changes:

- rename flag to `--environ-file-unsafe` (future can add `--environ-file`) due to node versions prior to node 22 consuming the `--env-file` https://github.com/babashka/nbb/issues/371
- addition of at least one test that exercises it in GHA somehow

kanaka committed:
test/runexamples: Reorg. Add non-continue fail test.

Restructure the cmd construction so that the bare and dockerized
versions have more similar arguments.

Add a third argument to `check` that takes additional options to pass to
dctest. Hoist --continue-on-error to be passed using this option and add
an additional faiure test without the option.

jonsmock commented on src/dctest/util.cljs:
```diff
@@ -8,9 +8,10 @@
     [dctest.expressions :as expr]
     [promesa.core :as P]
-    [viasat.util :refer [Eprintln fatal read-file]]
+    [viasat.util :refer [Eprintln fatal read-file exec-promise]]
```
not important: probably don't need exec-promise anymore :-)

kanaka replied:
I fixed that, thanks.

jonsmock reviewed (approved):
Yay! Looks awesome!
